api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/TaskTranslatorImpl.java

```diff
@@ -29,17 +27,40 @@ public Task toFhirResource(org.openmrs.module.fhir2.Task openmrsTask) {
 	public org.openmrs.module.fhir2.Task toOpenmrsType(Task fhirTask) {
 		org.openmrs.module.fhir2.Task openmrsTask = new org.openmrs.module.fhir2.Task();
 		
-		if (fhirTask != null && fhirTask.getStatus() != null && fhirTask.getIntent() != null) {
-			openmrsTask.setUuid(fhirTask.getId());
-			openmrsTask.setStatus(org.openmrs.module.fhir2.Task.TaskStatus.valueOf(fhirTask.getStatus().name()));
-			openmrsTask.setIntent(org.openmrs.module.fhir2.Task.TaskIntent.valueOf(fhirTask.getIntent().name()));
+		if (fhirTask != null) {
+			setOpenmrsTaskFields(openmrsTask, fhirTask);
 		}
 		
 		return openmrsTask;
 	}
 	
 	@Override
-	public org.openmrs.module.fhir2.Task toOpenmrsType(org.openmrs.module.fhir2.Task existingObject, Task resource) {
-		return null;
+	public org.openmrs.module.fhir2.Task toOpenmrsType(org.openmrs.module.fhir2.Task openmrsTask, Task fhirTask) {
+		if (fhirTask != null & openmrsTask != null) {
+			setOpenmrsTaskFields(openmrsTask, fhirTask);
+		}
+		
+		return openmrsTask;
+	}
+	
+	private void setFhirTaskFields(org.openmrs.module.fhir2.Task openmrsTask, Task fhirTask) {
+		fhirTask.setId(openmrsTask.getUuid());
+		if (openmrsTask.getStatus() != null) {
+			fhirTask.setStatus(Task.TaskStatus.valueOf(openmrsTask.getStatus().name()));
+		}
+		if (openmrsTask.getIntent() != null) {
+			fhirTask.setIntent(Task.TaskIntent.valueOf(openmrsTask.getIntent().name()));
+		}
+	}
+	
+	private void setOpenmrsTaskFields(org.openmrs.module.fhir2.Task openmrsTask, Task fhirTask) {
+		openmrsTask.setUuid(fhirTask.getId());
```

Comment:
I would suggest changing this to:

```java
if (openmrsTask.getUuid() == null) {
    openmrsTask.setUuid(fhirTask.getId());
}
```

Since we don't want to allow a FHIR client to change the UUID of an existing task. I should also rethink this in the other translators.